pkg/apis/org/v1/che_types.go

```diff
@@ -138,6 +138,16 @@ type CheClusterSpecServer struct {
 	// Disabling TLS may cause malfunction of some Che components.
 	// +optional
 	TlsSupport bool `json:"tlsSupport"`
+
+	// Use internal host names provided by cluster dns.
+	// It should be usefull to speed internal services traffic.
+	// Also it could avoid some proxy issues.
+	// Default value is true.
+	ServiceHostnames *bool `json:"serviceHostnames"`
+
+	// Internal cluster DNS service hostname suffix. It's a root dns domain. "cluster.local" by default for Kube-DNS.
+	ServiceHostnameSuffix string `json:"serviceHostnameSuffix"`
```

Comment:
We need a way to detect cluster domain.

Reply:
I applied draft method getClusterDomain

Reply:
`getClusterDomain` gets cname for default service "kubernetes.default.svc" and we parse dns domain from request value. I tested it works with minikube, minishift, Openshift 4. 

A bit more context for Openshift 4: dns in the openshift 4 was implemented with operator: https://github.com/openshift/cluster-dns-operator . In the README.md of this operator I see `cluster.local` is default and in the code looks like this dns domain even hardcoded: https://github.com/openshift/cluster-dns-operator/blob/924a03d47a42b42c942722eb41c673bab4bd8030/pkg/operator/controller/controller.go#L329
And you can't configure dns domain using custom operator's resource - there isn't  any field to do that. You have only status field: ClusterDomain which has value `cluster.local` . Also you can't change dns domain `cluster.local` using dns config map - operator will erase this changes: https://github.com/openshift/cluster-dns-operator/blob/0d46f0303675473ce5d022904ddaff23132bac97/pkg/operator/controller/controller_dns_configmap.go#L90 , because .cluster.local is desired value.
Also I see uses  default service resolution https://github.com/openshift/cluster-dns-operator/blob/924a03d47a42b42c942722eb41c673bab4bd8030/pkg/operator/controller/controller.go#L158 and https://github.com/openshift/cluster-dns-operator/blob/924a03d47a42b42c942722eb41c673bab4bd8030/pkg/operator/controller/controller.go#L172 That's why I think our detector should work fine on the Openshift 4.

Reply:
We decided to use services in format: service-name.namespace.svc